common/flagdproxy/flagdproxy.go

```diff
@@ -71,58 +76,99 @@ func (f *FlagdProxyHandler) Config() *FlagdProxyConfiguration {
 	return f.config
 }
 
-func (f *FlagdProxyHandler) createObject(ctx context.Context, obj client.Object) error {
-	return f.Client.Create(ctx, obj)
+func specDiffers(a, b client.Object) (bool, error) {
+	if a == nil || b == nil {
+		return false, fmt.Errorf("object is nil")
+	}
+
+	// Compare only spec based on the object type
+	switch a.(type) {
+	case *corev1.Service:
+		return !reflect.DeepEqual(a.(*corev1.Service).Spec, b.(*corev1.Service).Spec), nil
+	case *appsV1.Deployment:
+		return !reflect.DeepEqual(a.(*appsV1.Deployment).Spec, b.(*appsV1.Deployment).Spec), nil
+	case *policyv1.PodDisruptionBudget:
+		return !reflect.DeepEqual(a.(*policyv1.PodDisruptionBudget).Spec, b.(*policyv1.PodDisruptionBudget).Spec), nil
+	default:
+		return false, fmt.Errorf("unsupported object type")
+	}
 }
 
-func (f *FlagdProxyHandler) updateObject(ctx context.Context, obj client.Object) error {
-	return f.Client.Update(ctx, obj)
+// ensureFlagdProxyResource ensures that the given object is reconciled in the cluster. If the object does not exist, it will be created.
+func (f *FlagdProxyHandler) ensureFlagdProxyResource(ctx context.Context, obj client.Object) error {
+	if obj == nil {
+		return fmt.Errorf("object is nil")
+	}
+
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		var old = obj.DeepCopyObject().(client.Object)
+
```

Comment:
Since this is called potentially multiple times, maybe it is worth keeping the info logs to support debug activities.